dev/com.ibm.ws.container.service/src/com/ibm/ws/container/service/config/internal/WebFragmentsInfoImpl.java

```diff
@@ -61,13 +63,11 @@ class WebFragmentsInfoImpl implements WebFragmentsInfo {
         this.servletSpecLevel = servletSpecLevel;
 
         // Web app ...
-
         WebApp webApp = containerToAdapt.adapt(WebApp.class); // throws UnableToAdaptException
         if (webApp != null) {
             this.servletSchemaLevel = webApp.getVersion();
-            if ("6.0".equals(servletSchemaLevel) || "5.0".equals(servletSchemaLevel) || "4.0".equals(servletSchemaLevel) || "3.1".equals(servletSchemaLevel)
-                || "3.0".equals(servletSchemaLevel)
-                || "2.5".equals(servletSchemaLevel)) {
+            int servletSchemaLevelInt = PlatformVersion.getVersionInt(servletSchemaLevel);
+            if (Arrays.binarySearch(WebApp.METADATA_COMPLETE_SUPPORT, servletSchemaLevelInt) >= 0) {
```

Comment:
A HashSet in WebApp would be more efficient than binary search.  I would actually put this logic all in WebApp static method like if (WebApp.isMetadataCompleteSupported(servletSchemaLevel)) and not referencing public fields in WebApp class.